File: src/Pybind11Wraps/RK/RKFieldNames.py
from PYB11Generator import *

class RKFieldNames:

    rkOrders = PYB11readonly(static=True, returnpolicy="copy")
    rkCorrectionsBase = PYB11readonly(static=True, returnpolicy="copy")
    reproducingKernelBase = PYB11readonly(static=True, returnpolicy="copy")

    @PYB11static
    def rkCorrections(order = "const RKOrder"):
        return "std::string"

    @PYB11static
    def reproducingKernel(order = "const RKOrder"):
        return "std::string"

    @PYB11static
    def correctionOrder(x = "const std::string&"):
        return "RKOrder"

File: src/RK/RKFieldNames.cc
#include "RK/RKFieldNames.hh"

namespace Spheral {

const std::string RKFieldNames::rkOrders = "rkOrders";
const std::string RKFieldNames::rkCorrectionsBase = "rkCorrections_";
const std::string RKFieldNames::reproducingKernelBase = "reproducingKernel_";
const std::string RKFieldNames::rkCorrections(const RKOrder order)     { return RKFieldNames::rkCorrectionsBase + std::to_string(static_cast<int>(order)); }
const std::string RKFieldNames::reproducingKernel(const RKOrder order) { return RKFieldNames::reproducingKernelBase + std::to_string(static_cast<int>(order)); }
const RKOrder RKFieldNames::correctionOrder(const std::string& x)      { const auto i = x.find("_"); return static_cast<RKOrder>(std::stoi(x.substr(i+1))); }

}
